### ISSUE TITLE
Documents are not embed-able or viewable while processing

### ISSUE THREAD

**Issue by nathanstitt** [documentcloud/documentcloud#40]
We'd like to be more responsive to users who are embedding documents by allowing them to embed documents immediately after uploading.   This seems reasonable, since the image extraction doesn't take a prohibitive amount of time relative to the text extraction and indexing, and the images are uploaded to s3 as soon as they are extracted.

It's a relatively easy change UI change to do so, and I've completed the modifications in commit: https://github.com/nathanstitt/documentcloud/commit/4879670bc19471675ece6d543569cdef6ddc62ad

The resulting embed fails to load while the document is processing due to documents_controller returning 403 in the show method.  This traces back to Document#accessible checking if the access is set to PUBLIC, which it is not, it's PENDING.

TLDR; we need to create a way to indicate a document is processing but access should be viewable.

I see three avenues we can take to accomplish this:
- Add a new access level PROCESSING_PUBLIC (or some such).  I find this one distasteful as the change would be very invasive to the codebase, and wouldn't provide any future benefits.
- Covert the access levels into bitmasks.   By doing so a document could have multiple status's such as PROCESSING & PUBLIC.   This would keep us from having to clutter up the documents table with another column, but the downside is the same as the upside, it would be possible for a document to have two statuses.  PROCESSING & PUBLIC makes sense - PUBLIC & PRIVATE does not.
- Remove the processing access level and make it a boolean flag on the document model.
  This one is my favorite.  In my mind there are two concerns with the document.  
  - The access level, which controls who can view/modify it.
  -  Whether the document is being processed.
  
  A side benefit of this method is that documents would no longer lose their access level when an error occurs during processing.

Perhaps someone can think of a better method?
